### PR TITLE
Feature/t76197 create input component

### PR DESCRIPTION
**Pull request by t3o79**
This is first part of input component.
Here just create shell and some default style for it.
After designer change their input component alerts I will continue to work on this.

### CODE REVIEW

**Comment by aldinezi** [comment]
Please create this PR again. Dev branch was not in sync with master, that's why we have so many changes in this PR.
@alen-d you did not merged your commits into dev branch. 
@dika-pro please set-up some ground rules. Branches should be merged into dev first, and then into master.

File: src/components/InputBox.vue [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    </div>
+  </div>
+</template>
+
+<script>
+import { forEach } from 'lodash';
+import { inputTypeable, typeable } from '../mixins';
+
+export default {
+  name: 'InputBox',
```
Comment (dika-pro) [inline src/components/InputBox.vue]: Let's call it just `Input`?

We already use simple names (Button, Icon).

File: src/components/InputBox.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  props: [
+    'inputComponentType',
+    'inputIcon',
+    'inputLabelTitle',
+    'inputHelperText',
+    'inputDisable',
+    'inputRules',
```
Comment (dika-pro) [inline src/components/InputBox.vue]: You can omit `input` prefix.

File: src/components/InputBox.vue [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  ],
+  data() {
+    return {
+      inputValue: '',
```
Comment (dika-pro) [inline src/components/InputBox.vue]: Could we just use `value` here?
It just follows standard Vue model name (https://vuejs.org/v2/guide/components.html#Using-v-model-on-Components).

**Comment by t3o79** [comment]
What about emitting value of input on success? We need it. How will be that emit called?

File: src/components/InputBox.vue [hunk 4]
```diff
@@ -0,0 +1,74 @@
+<template>
+  <div class="input-wrapper"
+       :class="[inputTypeClass, typeClass, inputStatus]">
+    <div class="input-box"
+         :class="{'disabled': inputDisable}">
+      <div class="input">
+        <label :class="{'input-active': inputStatus !== 'default' || inputValue}">
+          {{inputLabelTitle}}
+        </label>
+        <input type="text"
```
Comment (aldinezi) [inline src/components/InputBox.vue]: Please apply lint rule here. Multiple attributes should all be separeted from tag and moved by only one indent. (just two spaces further than html tag)

I am not sure why there are no eslint warnings for this rule.

indent

Comment (t3o79) [reply src/components/InputBox.vue]: For me it is normal behavior, if the tag has attribute in that line he will fallow that attribute start position. If we remove attribute in that line, then the next line with two spaces.

So, what will do here? Left like this or remove all first attributes in new line?

Comment (aldinezi) [reply src/components/InputBox.vue]: This rule should be turned on. 
https://github.com/vuejs/eslint-plugin-vue/blob/master/docs/rules/max-attributes-per-line.md

You can leave it as you wish, I am just pointing out that it looks ugly, especially with odd number of spaces.

Comment (t3o79) [reply src/components/InputBox.vue]: I asked what will we do. That is not a problem I will change it.

**Comment by dika-pro** [comment]
> What about emitting value of input on success? We need it. How will be that emit called?
We agreed that this case will be handled by external lib (https://vuelidate.js.org/).

**Pull request closed without merging**